### PR TITLE
CRM-19562

### PR DESCRIPTION
**Pull request by seamuslee001**
* [CRM-19562](https://issues.civicrm.org/jira/browse/CRM-19562)

### CODE REVIEW

**Comment by seamuslee001** [comment]
ping @totten @eileenmcnaughton

File: tests/phpunit/CRM/Contact/BAO/QueryTest.php [hunk 1]
```diff
@@ -315,4 +315,32 @@ public function testGroupClause() {
...
+      "on_hold" => 1,
+      "display_name" => 1,
+      "preferred_mail_format" => 1,
+    );
+    $numberofContacts = 2;
+    $query = new CRM_Contact_BAO_Query($params, $returnProperties);
+    try {
+      $query->apiQuery($params, $returnProperties, NULL, NULL, 0, $numberofContacts);
+    }
+    catch (Exception $e) {
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Contact/BAO/QueryTest.php]: This test would still pass with no exception thrown. Try removing the try -catch & use an expectedException annotation https://thephp.cc/news/2016/02/questioning-phpunit-best-practices

Comment (seamuslee001) [inline tests/phpunit/CRM/Contact/BAO/QueryTest.php]: I tested and it fails on a DB Syntax error which is what was appearing before. (The messages are different so $assertEquals fails

Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Contact/BAO/QueryTest.php]: ah ok - then what you can do is add a return in the catch & then below it $this->markFail - so if it gets past the try-catch if fails

**Comment by eileenmcnaughton** [comment]
This adds validation to ensure contactIDs are all contactIDs

**Pull request merged**